src/base.iced

```diff
@@ -112,14 +112,28 @@ class Verifier
 
   #---------------
 
+  _check_json_formatting : (cb) ->
+    err = null
+    cb err
+
+  #---------------
+
   _check_json : (cb) ->
-    jsons = @payload
-    [e, @json] = katch (() -> JSON.parse jsons)
-    err = new Error "Couldn't parse JSON signed message: #{e.message}" if e?
-    if not err?
-      jsons = jsons.toString('utf8')
-      await @base._v_check {@json}, defer err
-    cb err, @json, jsons
+    json_str_buf = @payload
+    json_str_utf8 = json_str_buf.toString('utf8')
+    err = null
+    if not /^[\x20-\x7e]+$/.test json_str_utf8
+      err = new Error "All JSON proof characters must be in the ASCII set"
```

Comment:
Maybe this error should mention that non-ASCII characters can be included if escaped?